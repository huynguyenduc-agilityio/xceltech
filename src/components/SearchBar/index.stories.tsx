import type { Meta, StoryObj } from '@storybook/react';

// Components
import SearchBar from '.';

const meta = {
  title: 'Components/SearchBar',
  component: SearchBar,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],

  render: (args) => (
    <div className="flex justify-center w-[1000px] p-9 bg-blue-light">
      <SearchBar {...args} />
    </div>
  ),
} satisfies Meta<typeof SearchBar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    onChange: (value: string) => console.log(value),
  },
};
