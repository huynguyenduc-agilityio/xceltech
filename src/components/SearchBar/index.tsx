import { ChangeEvent, useCallback } from 'react';

// Icons
import { ChevronsUpDown } from 'lucide-react';
import { SearchIcon } from '@/icons';

// Components
import { Input, InputProps } from '../common';

type TSearchBarProps = Omit<InputProps, 'onChange'> & {
  onChange: (value: string) => void;
};

const SearchBar = ({ onChange, ...rest }: TSearchBarProps) => {
  const handleChangeValue = useCallback(
    (e: ChangeEvent<HTMLInputElement>): void => onChange(e.target.value),
    [onChange],
  );

  return (
    <div className="relative w-[721px] h-[69px]">
      <div className="absolute inset-0 flex items-center justify-center gap-3 w-[216px] h-full bg-primary text-white rounded-l-[15px]">
        <span className="text-lg font-bold">All Candidates</span>
        <ChevronsUpDown />
      </div>
      <Input
        type="text"
        placeholder="Search..."
        onChange={handleChangeValue}
        {...rest}
        className="bg-white h-full rounded-[15px] pl-[251px] pr-[90px] font-bold"
      />
      <SearchIcon className="absolute top-[25%] right-10" />
    </div>
  );
};

export default SearchBar;
