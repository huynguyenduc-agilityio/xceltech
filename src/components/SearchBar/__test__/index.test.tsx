import { fireEvent, render } from '@testing-library/react';

// Components
import SearchBar from '..';

const mockOnChange = jest.fn();

describe('SearchBar Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<SearchBar onChange={mockOnChange} />);

    expect(container).toMatchSnapshot();
  });

  it('should render input with placeholder', () => {
    const { getByPlaceholderText } = render(
      <SearchBar onChange={mockOnChange} />,
    );

    expect(getByPlaceholderText('Search...')).toBeInTheDocument();
  });

  it('Should call onChange function', () => {
    const { getByPlaceholderText } = render(
      <SearchBar onChange={mockOnChange} />,
    );

    const input = getByPlaceholderText('Search...');
    fireEvent.change(input, { target: { value: 'test' } });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledWith('test');
  });
});
