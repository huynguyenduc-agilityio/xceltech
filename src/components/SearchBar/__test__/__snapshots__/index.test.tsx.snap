// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should match snapshot 1`] = `
<div>
  <div
    class="relative w-[721px] h-[69px]"
  >
    <div
      class="absolute inset-0 flex items-center justify-center gap-3 w-[216px] h-full bg-primary text-white rounded-l-[15px]"
    >
      <span
        class="text-lg font-bold"
      >
        All Candidates
      </span>
      <svg
        class="lucide lucide-chevrons-up-down"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m7 15 5 5 5-5"
        />
        <path
          d="m7 9 5-5 5 5"
        />
      </svg>
    </div>
    <input
      autocomplete="off"
      class="font-productsans flex w-full px-5 py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-white h-full rounded-[15px] pl-[251px] pr-[90px] font-bold"
      data-testid="input-element"
      placeholder="Search..."
      type="text"
    />
    <svg
      class="absolute top-[25%] right-10"
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#a)"
      >
        <path
          d="m31.563 27.669-6.232-6.232A1.499 1.499 0 0 0 24.27 21H23.25A12.936 12.936 0 0 0 26 13c0-7.181-5.819-13-13-13S0 5.819 0 13s5.819 13 13 13c3.019 0 5.794-1.025 8-2.75v1.019c0 .4.156.781.438 1.062l6.23 6.232a1.494 1.494 0 0 0 2.12 0l1.768-1.77a1.507 1.507 0 0 0 .006-2.124ZM13 21c-4.419 0-8-3.575-8-8 0-4.419 3.575-8 8-8 4.419 0 8 3.575 8 8 0 4.419-3.575 8-8 8Z"
          fill="#C4C4C4"
        />
      </g>
      <defs>
        <clippath
          id="a"
        >
          <path
            d="M0 0h32v32H0z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
  </div>
</div>
`;
